src/app/components/Header/Header.js

```diff
@@ -1,36 +1,57 @@
 import React from 'react'
 import PropTypes from 'prop-types'
+import { withRouter } from 'react-router'
 
 import NetworkSwitcher from '../NetworkSwitcher'
 import MainNav from '../MainNav'
 
-import './Header.css'
+import home from '../../../img/home.svg'
 
-const Header = (props) => {
-  const { showMenu, setNetwork, selectedNetworkId, networks } = props
+import style from './Header.css'
+
+const getNavigation = props => {
+  const { account, history } = props
+  const loggedIn = account.address && account.wif
+  const isHomepage = history.location.pathname === '/'
+
+  if (!loggedIn && isHomepage) {
+    return
+  }
+
+  let navigation
+
+  if (loggedIn) {
+    navigation = <MainNav />
+  } else {
+    navigation = (
+      <button className={ style.mainNavigationNotLoggedIn } onClick={ () => history.push('/') }>
+        <img className={ style.mainNavigationNotLoggedInImg } src={ home } alt='house' />
+      </button>
+    )
+  }
+
+  return navigation
+}
+
+const Header = props => {
+  const { setNetwork, selectedNetworkId, networks } = props
+  const navigation = getNavigation(props)
 
   return (
-    <div styleName='header'>
-      <NetworkSwitcher setNetwork={ setNetwork } selectedNetworkId={ selectedNetworkId } networks={ networks } />
-      <div styleName='titleBar'>
-        { showMenu ? <div styleName='menuNavWrapper'><MainNav /></div> : null }
-        <div styleName='titleWrapper'>
-          <span styleName='titleName'>NeoLink</span>
-        </div>
+    <div className={ style.header }>
+      <div className={ style.menuNavWrapper }>{navigation}</div>
+      <div className={ style.headerTitle }>
+        <h1>NeoLink</h1>
       </div>
+      <NetworkSwitcher setNetwork={ setNetwork } selectedNetworkId={ selectedNetworkId } networks={ networks } />
     </div>
   )
 }
 
 Header.propTypes = {
-  showMenu: PropTypes.bool,
```

Comment:
I see you removing this from the component but 1) see comment above a bit, maybe we want this and 2) do you remove all instances of passing `showMenu` to this component.